charon/tests/cli.rs

```diff
@@ -111,3 +111,76 @@ fn charon_cargo_features() -> Result<()> {
         Ok(())
     })
 }
+
+fn rustup_install_target(target: &str) -> Result<()> {
+    // e.g. rustup target add riscv64gc-unknown-none-elf
+    let mut cmd = Command::new("rustup");
+    let output = cmd.args(["target", "add", target]).output()?;
+    ensure!(
+        output.status.success(),
+        "`rustup target add {target}` failed: {}",
+        String::from_utf8_lossy(&output.stderr)
+    );
+    Ok(())
+}
+
+#[test]
+#[ignore = "Network is inaccessible in nix"]
+fn charon_cargo_target() -> Result<()> {
+    let target = "riscv64gc-unknown-none-elf";
+    rustup_install_target(target)?;
+
+    let dir = "tests/cargo/multi-targets";
+    let fn_ = "pub fn multi_targets::";
+
+    #[cfg(target_family = "unix")]
```

Comment:
This looks incorrect: we're in a cross-compilation context; the test itself should be compiled and run on the host, only charon itself should be invoqued with the required target. The commands should be `charon cargo -- --target=blah`, no?

Reply:
Ah no I see, you're also testing the case where the host is different.